tests/runner/src/env.rs

```diff
@@ -178,27 +161,42 @@ impl Env {
             _ => panic!("Unexpected subcommand: {subcommand}"),
         }
 
-        let process = Command::new("./greptime")
+        let mut process = Command::new("./greptime")
             .current_dir(util::get_binary_dir("debug"))
             .args(args)
             .stdout(log_file)
             .spawn()
             .unwrap_or_else(|_| panic!("Failed to start the DB with subcommand {subcommand}"));
+
+        // check connection
+        let ip_addr = match subcommand {
+            "datanode" => DATANODE_ADDR,
+            "frontend" => SERVER_ADDR,
+            "metasrv" => METASRV_ADDR,
+            "standalone" => SERVER_ADDR,
+            _ => panic!("Unexpected subcommand: {subcommand}"),
+        };
+        if !util::check_port(ip_addr.parse().unwrap(), Duration::from_secs(10)).await {
+            Env::stop_server(&mut process).await;
+            panic!("Frontend doesn't up in 10 seconds, quit.")
+        }
+
         process
     }
 
     /// stop and restart the server process
     async fn restart_server(db: &GreptimeDB) {
         let mut server_process = db.server_process.lock().await;
         Env::stop_server(&mut server_process).await;
-        let new_server_process = Env::start_server("standalone", &db.ctx, false);
-        *server_process = new_server_process;
 
-        let is_up = util::check_port(SERVER_ADDR.parse().unwrap(), Duration::from_secs(2)).await;
-        if !is_up {
-            Env::stop_server(&mut server_process).await;
-            panic!("Server doesn't up in 10 seconds, quit.")
-        }
+        // check if the server is distributed or standalone
+        let subcommand = if db.frontend_process.is_some() {
```

Comment:
It's tricky, why not just add a field to `GreptimeDB` that represents if it's a standalone mode or not?